### PR TITLE
Make it possible to define the nominal_value of flows directly in the storage definition

### PR DESCRIPTION
**Pull request by uvchik**
The default value for the `nominal_input_capacity_ratio` (as well output) is now None.
 
```diff
self.nominal_input_capacity_ratio = kwargs.get(
- 'nominal_input_capacity_ratio', 0.2)
+ 'nominal_input_capacity_ratio', None)
```

I added some checks that raise errors if the storage is defined contradictory.

`nominal_value` im Flow AND `nominal_input_capacity_ratio` -> AttributeError
`nominal_value` im Flow OR `nominal_capacity`with investment -> AttributeError

Furthermore I wrote some tests and fixed existing ones to test the behaviour above.

TODO:
We could write a good error message to help users to understand what went wrong. At the moment
there are only drafts.

### CODE REVIEW

File: oemof/solph/components.py [hunk 1]
```diff
@@ -88,45 +90,30 @@ def __init__(self, *args, **kwargs):
         self.capacity_min = sequence(kwargs.get('capacity_min', 0))
         self.fixed_costs = kwargs.get('fixed_costs')
         self.investment = kwargs.get('investment')
+
         # Check investment
+        msg_no_nv = ("No nominal value allowed with investment object."
+                     "(DRAFT!) {0}")
```
Comment (ckaldemeyer) [inline oemof/solph/components.py]: What does DRAFT mean here?

File: oemof/solph/components.py [hunk 2]
```diff
@@ -88,45 +90,30 @@ def __init__(self, *args, **kwargs):
...
+            raise AttributeError(msg_no_nv.format('storage'))
+
+        # Check flows for nominal value
+        for flow in list(self.inputs.values()) + list(self.outputs.values()):
+            if self.investment and flow.nominal_value is not None:
+                raise AttributeError(msg_no_nv.format('flow'))
+            if (flow.nominal_value is not None and
+                    self.nominal_output_capacity_ratio is not None):
+                msg = ("Duplicate definition! (DRAFT!)"
+                       "fdsfgds")
```
Comment (ckaldemeyer) [inline oemof/solph/components.py]: See above. Is it meant to be adapted?

**Comment by ckaldemeyer** [comment]
> TODO:
> We could write a good error message to help users to understand what went wrong. At the moment
> there are only drafts.

Ahh... I should have read before ;-) @c-moeller : Do you have suggestions?

**Comment by c-moeller** [comment]
I have actually no idea what this feature does, sorry. Is there some more information or an issue which belongs?

**Comment by c-moeller** [comment]
Ah sorry, it's closed already... but anyway I'm interested in adaptions regarding storage.

**Comment by uvchik** [comment]
Have a look a t the first comment and at the "Files changed" tab. Ask if you still have questions.

**Pull request merged**